pkg/omni/repository/repository.go

```diff
@@ -0,0 +1,179 @@
+// Copyright 2018 Vulcanize
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package repository
+
+import (
+	"fmt"
+	"math/big"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/vulcanize/vulcanizedb/pkg/datastore/postgres"
+	"github.com/vulcanize/vulcanizedb/pkg/omni/contract"
+	"github.com/vulcanize/vulcanizedb/pkg/omni/types"
+)
+
+// Repository is used to
+type DataStore interface {
+	PersistEvents(info *contract.Contract) error
+}
+
+type dataStore struct {
+	*postgres.DB
+}
+
+func NewDataStore(db *postgres.DB) *dataStore {
+
+	return &dataStore{
+		DB: db,
+	}
+}
+
+// Creates a schema for the contract
+// Creates tables for the watched contract events
+// Persists converted event log data into these custom tables
+func (d *dataStore) PersistEvents(con *contract.Contract) error {
+
+	schemaExists, err := d.CheckForSchema(con.Name)
+	if err != nil {
+		return err
+	}
+
+	if !schemaExists {
+		err = d.CreateContractSchema(con.Name)
+		if err != nil {
+			return err
+		}
+	}
+
+	for eventName := range con.Filters {
+		event := con.Events[eventName]
```

Comment:
is this `eventName` a types.Filter? would we need to call `.Name` on it to get the name of the filter?

Reply:
con.Filters is a map of contract names to their filters so it should be good!